tfx/experimental/dummy_executor.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from typing import Any, Dict, List, Text
+from tfx import types
+from tfx.components.base import base_executor
+from tfx.types.artifact import Artifact
+
+import json
+import absl
+import os
+import filecmp
+from distutils.dir_util import copy_tree
+
+class DummyExecutor(base_executor.BaseExecutor):
+  def _compare_contents(self, uri: Text, expected_uri: Text):
+    """
+    recursively compare two directories,
+    Files are equal if names and contents are equal.
+    """
+    dirs_cmp = filecmp.dircmp(uri, expected_uri)
+    absl.logging.info("uri: %s", uri)
+    absl.logging.info("expected_uri: %s", expected_uri)
+    if len(dirs_cmp.left_only) > 0 or len(dirs_cmp.right_only) > 0 or \
+      len(dirs_cmp.funny_files) > 0:
+      return False
+
+    (_, mismatch, errors) = filecmp.cmpfiles(
+        uri, expected_uri, dirs_cmp.common_files, shallow=False)
+
+    if len(mismatch) > 0 or len(errors) > 0:
```

Comment:
nit: You can use the fact that empty list is False. For example, `if mismatch or errors:`

See [Python style guide](http://google.github.io/styleguide/pyguide.html#214-truefalse-evaluations) for more explanation.